modules/authorization-xacml/src/main/java/org/opencastproject/authorization/xacml/XACMLAuthorizationService.java

```diff
@@ -110,6 +116,11 @@ public void modified(Map<String, Object> config) {
     // updated() will handle the configuration update.
   }
 
+  @Reference(cardinality = ReferenceCardinality.OPTIONAL)
+  public void setMediaPackageSerializer(MediaPackageSerializer serializer) {
```

Comment:
Yes, there are several `MediaPackageSerializer`. Yet, all of these are managed by `ChainingMediaPackageSerializer`, and the `encodeURI`, `decodeURI` method will processed by all serializer one by one.
Serializers will check if the input URI should be processed by them. And `PresignedUrlMediaPackageSerializer` will only handle URI match S3 config.

Reply:
I think this can currently break in two ways, the first of which we are experiencing right now at SWITCH:

- This will cause workspace ACL URLs to be signed with a Stream Security policy since the signing provider only has a base URL to go on (in this case the admin URL) and that matches here, but the signing provider wasn't previously invoked in this context. This will later cause workspace and WFR URLs not to match, which will cause unnecessary reloading of files.
- As Greg said, there are multiple services providing the MediaPackageSerializer interface available via OSGI, so I don't see how you can guarantee to get the right one (or the one chaining up the rest). But then, we seem to do just that in other classes as well, and I don't understand how that works? Luck??